src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -81,28 +110,216 @@ void VulnerabilityScannerFacade::start(
         auto scanOrchestrator = std::make_shared<ScanOrchestrator>(
             m_indexerConnector, m_databaseFeedManager, m_reportSocketClient, m_internalMutex);
 
+        m_rebootThread = std::thread(
```

Comment:
We shouldn't process the reset of the vulnerabilities in another thread but in the main one before all the rest of the actions begin.

Reply:
I think this idea is contrary to what we discussed with @Dwordcito  in the last meeting since the approach was to generate a thread that would do a fetch of the global DB and an insert of the packages of each agent.

Reply:
If the `m_rebootThread` reads `shouldClean`, I understand that its job is to remove everything when the module was turned off.
The clean operation must be run before everything starts, the fetch and scan can be done at the same time.

Maybe I made a mistake and the cleaning is being performed before, but if this is the case, please use more clear variable names.

Reply:
Fix it!